src/extended_json.ts

```diff
@@ -70,16 +72,22 @@ const keysToCodecs = {
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 function deserializeValue(value: any, options: EJSONOptions = {}) {
   if (typeof value === 'number') {
+    const in32BitRange = value <= BSON_INT32_MAX && value >= BSON_INT32_MIN;
+    const in64BitRange = value <= BSON_INT64_MAX && value >= BSON_INT64_MIN;
+    if (options.useBigInt64 && !in32BitRange && in64BitRange) {
+      return BigInt(value);
```

Comment:
We've confirmed that the value is typeof number but not that it is an integer, won't this prevent falling through to the Double case?

Reply:
Good catch here. Seems like this should have been caught by a test.